### DISCUSSION TITLE
Weeknumber starting on Sunday

### DISCUSSION THREAD

**Discussion by RonnyLam** [joeycastillo/Sensor-Watch#181] [Q&A]
Hi,

It seems that the weeknumber is starting on Sunday. However, in Europ we start our weeknumbers on Monday. I have been looking at line 41 in watch_utility.c where I think the weeknumber is constructed. I don't understand the formula there very well, but I think I have to remove the last +1. Could someone confirm, or point me in the right direction to have weeknumbers start on Monday?

**Comment by RonnyLam** [comment]
I tested this myself and this is not the solution. This substracts the weekdaynumber by one day.
i.e., I tested this on Sunday January 15th. The weeknumber says, without changes, 03. But for Europe it should be 02. Implementing my change changes it indeed to week 02, but also to Saterday.

**Reply by RonnyLam** [reply]
I solved it by replacing line 49 in watch_utility.c with
    weekday = ((watch_utility_get_iso8601_weekday_number(year, month, day) + 5) % 7) + 1;
as per https://en.wikipedia.org/wiki/Zeller%27s_congruence for the ISO week date. 
I don't want to make this a PR (yet) as I am not sure what most people are using. At least in Europe Sunday the 29th of January 2023 is still week 4. Monday is week 5.

**Reply by RonnyLam** [reply]
@josecastillo what do you think?

**Reply by joeycastillo** [reply]
This sounds to me like it could be a good candidate for a preference in movement preferences, if I could figure out what the preference is. It sounds like more than week numbers to me; would it make sense to have a `use_iso_8601_calendar` option?

**Reply by RonnyLam** [reply]
That sounds like a viable option. It could be put in ```movement_config.h``` and tested for in ```watch_utility.c```. Or if you want a preference in the watch it would be something like "week starts on Monday" or "week starts on Sunday".

**Reply by joeycastillo** [reply]
For the moment though you could just implement this as a #define or a constant in the week number face, and have logic that does one thing or the other depending on its value.

**Reply by MikaelWeidenhielm** [reply]
This is exactly what i'm after. I also found the constant that needs changing. However i'm not able to build the firmware because of tons of errors being thrown. Could this be implemented in the preferences face in some way, for users not having the knowledge of doing it manually in the code? 🙏
